### ISSUE TITLE
代码可以更全一点吗

### ISSUE THREAD

**Issue by sounds** [lanwuwei/Subword-PWIM#1]
修改了一下，train完，发现模型好像没有保存。。
也没有test。

**Comment by lanwuwei** [comment]
I think these two files can answer your question: [train_quora.py](https://github.com/lanwuwei/SPM_toolkit/blob/master/SSE/train_quora.py) and [test_quora.py](https://github.com/lanwuwei/SPM_toolkit/blob/master/SSE/test_quora.py).

**Comment by sounds** [comment]
Thank you for the replay.
I see the embedding dict is saved  in the model subword-PWIM.  SO we should put all words embedding into the dict? ( The training word is a subset of vocab)
If the sentences we want to predict have new words(like fasttext embedding), how should we deal with it ?

**Comment by lanwuwei** [comment]
Yes, we should maintain a dictionary (or other data structure) to map words/subwords into embeddings. 
If we encounter new word/subword, we just use 'OOV' to replace it.

**Comment by sounds** [comment]
Initalizing out-of-vocabulary word vectors to zero vectors seems the  same as replacing the new word to ‘OOV’.
I use this model to do paraphrase identification in Chinese short sentences(query) , but there is two problems, do you have some advises:
1. if the terms of sentA  are almost  as same as the terms in sentB( sentB has more terms), the result is always 1.
2. if the sentence is very short,  the result seems not very good.

**Comment by lanwuwei** [comment]
Hi, these are two existing problems for PWIM. If you can solve it, you can definitely publish a paper~ My suggestions is that you can consider [soft attention](https://github.com/lanwuwei/SPM_toolkit) in DecAtt and ESIM or intra-attention in [IWAN](https://aclanthology.info/pdf/D/D17/D17-1122.pdf), since the 'hard attention' in PWIM is not a smart idea.
